ee/tabby-webserver/src/service/repository.rs

```diff
@@ -1,13 +1,59 @@
+use std::path::Path;
+
+use anyhow::anyhow;
 use async_trait::async_trait;
+use futures::StreamExt;
 use juniper::ID;
+use tabby_common::{config::RepositoryConfig, SourceFile};
 use tabby_db::DbConn;
 
 use super::{graphql_pagination_to_filter, AsID, AsRowid};
 use crate::schema::{
-    repository::{Repository, RepositoryService},
+    repository::{FileEntry, Repository, RepositoryMeta, RepositoryService},
     Result,
 };
 
+fn match_glob(path: &str, glob: &str) -> bool {
+    let glob = glob.to_lowercase();
+    let mut path = &*path.to_lowercase();
+    // Current behavior: Find each "word" (any substring separated by whitespace) appearing in the same order
```

Comment:
@wsxiaoys Tried using `ignore` but couldn't get it behaving as I wanted, so I switched to `async_walkdir` with my own glob filter instead. This comment explains its behavior, and I can show some screenshots to demonstrate how it works:
![image](https://github.com/TabbyML/tabby/assets/14198267/804b7100-800b-4bff-b05b-f31899998245)
![image](https://github.com/TabbyML/tabby/assets/14198267/a669374c-fcdc-46ae-8aef-21603c2d4d17)
![image](https://github.com/TabbyML/tabby/assets/14198267/4aaf26a9-c8b5-42c4-b592-0bd5ff82e6f2)

Reply:
cc @liangfung

Reply:
1. I don't understand why it cannot be using ignore - since you're implementing your own glob match anyway.
2. Please also returns directories.

Reply:
1. ignore does not allow for custom globs for matching, only ignoring files as far as I can tell
2. ignore is not async, and so would block the entire server's async runtime while it is running a search

Reply:
1. As said - glob is implemented as customized filter anyway
2. tokio manage its requests in multi threads async runtime. Thus won’t block